### PR TITLE
Fixes #20137 - hide ssh keys for new users

### PR DESCRIPTION
**Pull request by ares**
This might be a candidate for 1.15.z

### CODE REVIEW

File: app/views/ssh_keys/_ssh_keys_tab.html.erb [hunk 1]
```diff
@@ -27,10 +27,14 @@
...
               <%= _("The keys can be used in provisioning templates and are also available for configuration management tools managed by Foreman.") %><br />
              </p>
-             <% if authorized_for(hash_for_new_user_ssh_key_path(:user_id => @user)) %>
+             <% if @user.persisted? && authorized_for(hash_for_new_user_ssh_key_path(:user_id => @user)) %>
                <div class="blank-slate-pf-main-action">
                  <%= link_to(_('Create SSH key'), hash_for_new_user_ssh_key_path(:user_id => @user), :class => "btn btn-success btn-lg" ) %>
                </div>
+             <% else %>
+               <p>
+                 <%= _("In order to create the SSH key, the user must be saved first. A 'create_ssh_keys' is required too.") %>
```
Comment (dLobatog) [inline app/views/ssh_keys/_ssh_keys_tab.html.erb]: I think the message got cut by mistake, you meant "A `create_ssh_keys` permission is required"?

Comment (dLobatog) [reply app/views/ssh_keys/_ssh_keys_tab.html.erb]: Also this immediately reminded me of the same thing in Roles `https://github.com/theforeman/foreman/blob/develop/app/views/roles/_form.html.erb#L46`, mind to also use `alert :header => '', :class => 'alert-warning'` for coherence?

**Review by dLobatog — CHANGES_REQUESTED** [review]
Thanks @ares , I agree this is 1.15.2, double check the warning message please.

**Comment by ares** [comment]
Both good catches, I updated the message and styled a warning. Please take another look.

**Review by tbrisker — APPROVED** [review]

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @ares  & @tbrisker !

**Pull request merged**